### DISCUSSION TITLE
About the recall , use bootcamp

### DISCUSSION THREAD

**Discussion by liuyichen2013** [milvus-io/milvus#18713] [Q&A and General discussion]
milvus version：2.1.1
test  scripts：https://github.com/milvus-io/bootcamp/tree/master/benchmark_test/scripts

I use the dataset Milvus_ Sift1m inserts a million vectors, and the index value should be 0-99999. Then use benchmark to perform Milvus recall test. No matter how you modify npore and TOPK, the result is 100%.

During retrieval, the index corresponding to the returned result must be between 0-99999, and then intersection with the ground_ truth_ 1m . I found the ground_ truth_ 1m seems to contain all the indexes (0-99999), that is, all the search results can be found in ground_ truth_ 1m, so the recall rate is 100%. 
I think ground_ truth_1m should not contain all index values. I wonder if my understanding is correct

**Comment by yhmo** [comment]
According to this page: https://github.com/milvus-io/bootcamp/blob/master/benchmark_test/lab1_sift1b_1m.md
The **ground_ truth_1m.txt** is downloaded from here: https://drive.google.com/file/d/1vBP9mKu5oxyognHtOBBRtLvyPvo8cCp0/view?usp=sharing
The **query_data.npy** is downloaded from here: https://drive.google.com/file/d/17jPDk93PQsB5yGh1J1YD9N7X8jvPEUQL/view?usp=sharing

The page gives a description for the two files:
**Unzip the query data to get the query_data folder that contains query.npy, which contains 10,000 vectors to query.
Unzip the ground truth data to get the gnd folder with ground_truth_1M.txt, which contains the locations of top 1000 similar vectors in the query data.**

So, we can know the query_data.npy contains 10000 vectors to do the benchmark, the ground_ truth_1m.txt is a ANN search result for the 10000 vectors with topk=1000.

**Reply by yhmo** [reply]
> So, ground_ truth_ 1m.txt contains 10 million IDS, which seems to contain all IDS. No matter what the search ID is, on the ground_ truth_1m.txt Can be found , and the recall rates are 100%

Yes, the ground_ truth_ 1m.txt contains 10 million IDS.
But when you compare the recall rate, you should not compare the 10m ids.

Let's give a name for each of the 10000 vectors to query:  "v1", "v2", "v3" .... "v10000".

The ground_ truth_ 1m.txt records top1000 for each vector:
Line 1 to line 1000 for "v1"
Line 1001 to line 2000 for "v2"
Line 2001 to line 3000 for "v3"
......

Now you pick "v2" to do a search with topk=10, you get 10 ids from milvus, and now you want to know the recall rate.
Then, compare the 10 ids with the line 1001 to line 1011. If all the 10 ids are matched, then we say the recall rate is 100%.

**Comment by liuyichen2013** [comment]
Sorry, I misunderstood，thanks
